prov/bgq/include/rdma/bgq/fi_bgq_flight_recorder.h

```diff
@@ -0,0 +1,128 @@
+
+
+#ifndef PROV_BGQ_INCLUDE_FLIGHT_RECORDER_H
+#define PROV_BGQ_INCLUDE_FLIGHT_RECORDER_H
+
+//#define FLIGHT_RECORDER_ENABLE
+#include "rdma/bgq/fi_bgq_spi.h"
+
+#define FLIGHT_RECORDER_ENTRY_STRLEN (256-8-2)
+#define FLIGHT_RECORDER_ENTRY_COUNT (1024)
+#define FLIGHT_RECORDER_INTERVAL (1024*1024*1024)
+
+struct flight_recorder_entry {
+	uint64_t	timebase;
+	uint16_t	event;
+	//uint16_t	line;
+	char		str[FLIGHT_RECORDER_ENTRY_STRLEN];
+};
+
+struct flight_recorder {
+#ifdef FLIGHT_RECORDER_ENABLE
+	uint64_t	last_dump;
+	unsigned	count;
+	uint32_t	rank;
+	int		thread;
+	uint8_t		pad[64-8-4-4-4];
+	struct flight_recorder_entry	entry[FLIGHT_RECORDER_ENTRY_COUNT];
+#endif
```

Comment:
It might be better to just not include it at all (per the whole "don't commit commented-out code" philosophy).

Something we've talked about internally here at Cisco but have never had the time to do would be to add proper support for [LTTNG](http://www.lttng.org/).